pkg/repo/repotest/server.go

```diff
@@ -185,31 +169,39 @@ func (srv *OCIServer) Run(t *testing.T, opts ...OCIServerOpt) {
 		t.Fatal("error removing chart before push")
 	}
 
-	err = registryClient.SaveChart(ch, ref)
+	// save it back to disk..
+	absPath, err := chartutil.Save(ch, srv.Dir)
+	if err != nil {
+		t.Fatal("could not create chart archive")
+	}
+
+	// load it into memory...
+	contentBytes, err := ioutil.ReadFile(absPath)
 	if err != nil {
-		t.Fatal("error saving chart")
+		t.Fatal("could not load chart into memory")
 	}
 
-	err = registryClient.PushChart(ref)
+	_, err = registryClient.Push(contentBytes, ref)
 	if err != nil {
-		t.Fatal("error pushing chart")
+		t.Fatalf("error pushing dependent chart: %s", err)
 	}
 
 	if cfg.DependingChart != nil {
 		c := cfg.DependingChart
-		dependingRef, err := ociRegistry.ParseReference(fmt.Sprintf("%s/u/ocitestuser/oci-depending-chart:1.2.3", srv.RegistryURL))
-		if err != nil {
-			t.Fatal("error parsing reference for depending chart reference")
-		}
+		dependingRef := fmt.Sprintf("%s/u/ocitestuser/%s:%s",
+			srv.RegistryURL, c.Metadata.Name, c.Metadata.Version)
 
-		err = registryClient.SaveChart(c, dependingRef)
+		// load it into memory...
+		absPath := filepath.Join(srv.Dir,
+			fmt.Sprintf("%s-%s.tgz", c.Metadata.Name, c.Metadata.Version))
+		contentBytes, err := ioutil.ReadFile(absPath)
 		if err != nil {
-			t.Fatal("error saving depending chart")
+			t.Fatal("could not load chart into memory")
 		}
 
-		err = registryClient.PushChart(dependingRef)
+		_, err = registryClient.Push(contentBytes, dependingRef)
```

Comment:
do we want to throw away the `PushResult`?   Perhaps a debug log?

Reply:
Resolved in 30a8ce59861a4d9fc935d8e1dd53664208876621